ArmPkg/Drivers/CpuDxe/CpuDxe.c

```diff
@@ -291,13 +291,60 @@ RemapUnusedMemoryNx (
       ArmSetMemoryAttributes (
         MemoryMapEntry->PhysicalStart,
         EFI_PAGES_TO_SIZE (MemoryMapEntry->NumberOfPages),
-        EFI_MEMORY_XP,
-        EFI_MEMORY_XP
+        0,
+        0,
+        TRUE
         );
     }
 
     MemoryMapEntry = NEXT_MEMORY_DESCRIPTOR (MemoryMapEntry, DescriptorSize);
   }
+
+  FreePool (MemoryMap);
+
+  MemoryMapSize = 0;
+  MemoryMap     = NULL;
+
+  Status = gBS->GetMemoryMap (
+                  &MemoryMapSize,
+                  MemoryMap,
+                  &MapKey,
+                  &DescriptorSize,
+                  &DescriptorVersion
+                  );
+  ASSERT (Status == EFI_BUFFER_TOO_SMALL);
+  do {
+    MemoryMap = (EFI_MEMORY_DESCRIPTOR *)AllocatePool (MemoryMapSize);
+    ASSERT (MemoryMap != NULL);

```

Comment:
It looks like a `MemoryMap` null pointer here will result in an error being returned from `GetMemoryMap()` (`EFI_INVALID_PARAMETER`) and then an attempt to call `FreePool()` on it when asserts are disabled.